packages/ui/src/CheckBox.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+import { useEffect, useState } from "react";
+import { TrackCardProps } from "./TrackCard";
+import { checkMarkAsCompleted, updateMarkAsCompleted } from "../../../apps/web/components/utils";
+
+export const CheckBox = ({ track }: { track: TrackCardProps }) => {
+  const [isChecked, setIsChecked] = useState(false);
+  useEffect(() => {
+    fetchCheckboxState();
+  }, []);
+
+  const fetchCheckboxState = async () => {
+    try {
+      const response = (await checkMarkAsCompleted(track.id)) || false;
```

Comment:
This would lead to a lot of API calls
This should happen in a single call to get the value of `checked`